src/za/ac/sun/cs/green/expr/Operation.java

```diff
@@ -124,7 +124,7 @@ public Fix getFix() {
 
 	public static final Expression TRUE = new Operation(Operation.Operator.EQ, ZERO, ZERO);
 
-	private final Operator operator;
+	private Operator operator;
```

Comment:
This is a very dirty way of doing things. I don't believe you need to make changes to Operation like this in order to achieve constant propagation. This is a pretty big change to the existing program, you really need to motivate why it is necessary to make this change.